src/Microsoft.Identity.Web/AppServicesAuth/AppServicesAuthenticationHandler.cs

```diff
@@ -15,20 +15,19 @@ namespace Microsoft.Identity.Web
     /// </summary>
     public class AppServicesAuthenticationHandler : AuthenticationHandler<AppServicesAuthenticationOptions>
     {
-        /// <summary>
-        /// Constructor for the AppServiceAuthenticationHandler.
-        /// Note the parameters are required by the base class.
-        /// </summary>
-        /// <param name="options">App service authentication options.</param>
-        /// <param name="logger">Logger factory.</param>
-        /// <param name="encoder">URL encoder.</param>
-        /// <param name="clock">System clock.</param>
+        /// <inheritdoc/>
+        ///<remarks>Note the parameters are required by the base class.</remarks>
         public AppServicesAuthenticationHandler(
               IOptionsMonitor<AppServicesAuthenticationOptions> options,
               ILoggerFactory logger,
+#if NET8_0_OR_GREATER
+              UrlEncoder encoder)
+              : base(options, logger, encoder)
+#else
               UrlEncoder encoder,
               ISystemClock clock)
```

Comment:
Oh they introduced a breaking change in the auth handlers?

Reply:
yes, systemClock is obsolete and replaced by TimeProvider. The base constructor does not take a TimeProvider, though; the systemClock arg is simply removed.